app/pdfs/donation_pdf.rb

```diff
@@ -0,0 +1,196 @@
+# Configures a Prawn PDF template for generating Donation receipts
+class DonationPdf
+  include Prawn::View
+  include ItemsHelper
+
+  class DonorInfo
+    attr_reader :name, :address, :email
+
+    def initialize(donation = nil)
+      if donation.nil?
+        @name = @address = @email = nil
+        return
+      end
+      case donation.source
+      when "Donation Site"
+        @name = donation.donation_site.name
+        @address = donation.donation_site.address
+        @email = donation.donation_site.email
+      when "Manufacturer"
+        @name = donation.manufacturer.name
+        @address = nil
+        @email = nil
+      when "Product Drive"
+        @name = donation.product_drive_participant.business_name
+        @address = donation.product_drive_participant.address
+        @email = donation.product_drive_participant.email
+      when "Misc. Donation"
+        @name = "Misc. Donation"
+        @address = nil
+        @email = nil
```

Comment:
Can we extend or use the `details` method of the `Donation` class? I feel like this is more generally useful than just inside the PDF.

Reply:
On a couple of the sources, the information returned by details is different from what seems to make sense for contact info.  For example, on product drives, the drive name is returned, but seems the participant would make more sense for the contact info.

Maybe have details return DonorInfo, with existing info returned as a details variable (donor.details)?

Or should what is currently displayed in the PDF for product drive donations and misc donations be changed?

Reply:
Heh let's not go down a rabbit hole - I'm fine with leaving it, I just wanted to explore if there would be utility in having the information shared. @cielf are the discrepancies good and we should keep this separate? Or do we want to define it centrally?

Reply:
I think it's good as it is now -- I could see merit in having the details show *both* the participant and the drive, but that wouldn't give us the right thing for the "donation from",  which should just be the participant, unless we hear differently from the banks.